common/consistenthash/consistenthash.go

```diff
@@ -22,7 +22,19 @@ func (r *Ring) search(key string) int {
     // YOUR CODE GOES HERE //
     /////////////////////////
 
-    return 0
+    // Generating the position on the ring
+    hashValue := hashId(key)
+    idx := 0
+    // Walking clockwise to find the first node with
+    // a position larger than the item's position
+    for _, node := range r.Nodes {
+        if node.HashId > hashValue {
```

Comment:
Mas se a gente for olhar especificamente o papel da função search, incluindo o comentário acima dela: 

```go
// search will find the index of the node that is responsible for the range that
// includes the hashed value of key.
func (r *Ring) search(key string) int {
```

Você consideraria esse comportamento correto? Provavelmente o seu código vai ter um erro no corretor automático. Ele simplesmente vai testar a API esperada da biblioteca consistenthash, e não se o código está funcionando corretamente.